### ISSUE TITLE
[$75] Screen Reader: Group Chat: The button 'Start a chat, refer a friend' is announced as 'referral'

### ISSUE THREAD

**Issue by rushatgabhane** [rushatgabhane/exfy#26]
Action Performed:
Precondition: User is logged in

Using VoiceOver on iOS or TalkBack on Android, open the app
Swipe to focus the Create tab and double-tap to activate it
Swipe to focus the Start chat button and double-tap to activate it
Swipe to focus the "Start a chat, refer a friend" button
Note how the element is displayed and announced.
Expected Result:
The button should be announced with its displayed text, e.g.:

'Start a chat, refer a friend,'
Actual Result:
The button 'Start a chat, refer a friend' is announced as 'referral'.

Area issue was found in:
Group chat

Failed WCAG checkpoints
WCAG 2.5.3 Label in Name
https://www.w3.org/WAI/WCAG22/Understanding/label-in-name

User impact:
Speech-input users can directly activate controls on a page with fewer surprising changes of focus. Text-to-speech users will have a better experience because the labels that they hear match the visible text labels that they see on the screen.

Suggested resolution:
Update the accessibilityLabel to match the visible text and ensure it is announced with the visible text:

'Start a chat, refer a friend'
https://appt.org/en/docs/react-native/samples/accessibility-label
https://reactnative.dev/docs/accessibility#accessibilitylabel
Environment:
Apple iPhone 14 Pro - iOS 18.7.1, Wi-Fi - Native (No mobile browser)
Samsung Galaxy Note10 Plus - Android 12, H0T Mobile - Native (No mobile browser)

Screenshots/Videos

<img width="1179" height="2438" alt="Image" src="https://github.com/user-attachments/assets/f8d7a863-3f6b-4be5-8c1a-e6e94b125f6b" />

https://github.com/user-attachments/assets/c6fa6680-f4cd-48d6-b675-b78fefc9cc5c

Version Number: Production
Reproducible in staging?: NA
Reproducible in production?: Yes
Email or phone of affected tester (no customers): No
Logs: https://stackoverflow.com/c/expensify/questions/4856 NA
Notes/Photos/Videos: Any additional supporting documentation Yes
Expensify/Expensify Issue URL: https://platform.applause.com/company/12102/products/36628/community-issues/7007944
Issue reported by: NA
Slack conversation: NA
